### PR TITLE
Enabled KWD demo on iMX8M arch

### PR DESCRIPTION
**Pull request by cristinailie**
- Created KWD topology for iMX8M (adapting the KWD topology from APL arch)

- Enabled detect_test component default compilation for iMX arch (by enabling CONFIG_SAMPLE_KEYPHRASE in defconfig file)

### CODE REVIEW

**Comment by sofci** [comment]
Can one of the admins verify this patch?
> reply **test this please** to run this test once.

**Comment by gkbldcig** [comment]
Can one of the admins verify this patch?

File: tools/topology/CMakeLists.txt [hunk 1]
```diff
@@ -119,6 +119,7 @@ set(TPLGS
 	"sof-imx8-wm8960-mixer\;sof-imx8-wm8960-mixer"
```
Comment (dbaluta) [inline tools/topology/CMakeLists.txt:119]: Mostly looking good. We need to adapt the commit messages to reflect the coding style of the project.

First the commit subject should use some tags relevant for the subsystem being updated. E.g `topology`. Also, use imperative mood in the commit subject. E.g: `Create` instead of `Created`

So, the commit message should look like this:

```
topology: Create a KWD topology for i.MX8MP platform

Adapted the APL KWD topology to iMX8MP by switching to
SCHEDULE_TIME_DOMAIN_DMA on capture pipeline and
creating a no LP capture file for KFBM component.
```

File: tools/topology/sof-imx8mp-keyword-detect.m4 [hunk 2]
```diff
@@ -0,0 +1,96 @@
+#
+# Topology for i.MX8 with wm8960 codec for
```
Comment (iuliana-prodan) [inline tools/topology/sof-imx8mp-keyword-detect.m4:2]: s/i.MX8/i.MX8MP

File: src/arch/xtensa/configs/imx8_defconfig [hunk 3]
```diff
@@ -2,3 +2,4 @@ CONFIG_IMX8=y
 CONFIG_HAVE_AGENT=n
 CONFIG_FORMAT_CONVERT_HIFI3=n
 CONFIG_KPB_FORCE_COPY_TYPE_NORMAL=n
+CONFIG_SAMPLE_KEYPHRASE=y
```
[the remark this answers is not in the document]
Comment (dbaluta) [reply src/arch/xtensa/configs/imx8_defconfig:5]: @cristinailie I don't think adding an option to defconfig is needed because once you have this:

```
        config SAMPLE_KEYPHRASE
»       »       depends on CAVS || IMX                                                                                                                                                                     
»               bool "Keyphrase test com
```

the CONFIG_SAMPLE_KEYPHRASE will be automatically selected.

File: tools/topology/sof-imx8mp-keyword-detect.m4 [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+dnl PIPELINE_ADD(pipeline,
+dnl     pipe id, max channels, format,
+dnl     period, priority, core,
+dnl     sched_comp, time_domain,
+dnl     pcm_min_rate, pcm_max_rate, pipeline_rate)
+PIPELINE_ADD(sof/pipe-detect.m4,
+	2, 2, s16le,
+	KWD_PIPE_SCH_DEADLINE_US, 1, 0,
+	PIPELINE_SCHED_COMP_1,
+	SCHEDULE_TIME_DOMAIN_TIMER,
```
Comment (ViorelSuman) [inline tools/topology/sof-imx8mp-keyword-detect.m4:71]: I thought we need SCHEDULE_TIME_DOMAIN_DMA here, right ?

Comment (paulstelian97) [reply tools/topology/sof-imx8mp-keyword-detect.m4:71]: We need DMA for the pipeline that includes the DAI but timer for the second one.

Comment (ViorelSuman) [reply tools/topology/sof-imx8mp-keyword-detect.m4:71]: > We need DMA for the pipeline that includes the DAI but timer for the second one.

It's scheduled on PIPELINE_SCHED_COMP_1 which is SAI3 and DMA driven, isn't it ? So why TIMER ?

Comment (paulstelian97) [reply tools/topology/sof-imx8mp-keyword-detect.m4:71]: @ViorelSuman We can try putting both on DMA, but I believe there were issues (in the sense that it wouldn't run) when doing that. @cristinailie Do you remember if we have actually tried that?

Comment (cristinailie) [reply tools/topology/sof-imx8mp-keyword-detect.m4:71]: I just tried with both on DMA and I cannot see any difference in behavior.

File: src/samples/audio/Kconfig [hunk 5]
```diff
@@ -11,7 +11,7 @@ menu "Audio component samples"
 	                Select for test smart amplifier component
 
         config SAMPLE_KEYPHRASE
-		depends on CAVS
+		depends on CAVS || IMX
```
Comment (paulstelian97) [inline src/samples/audio/Kconfig:14]: Commit message is slightly inaccurate. You are adding a dependence on the IMX platform, not architecture.

I'd rephrase the commit message as "Add IMX platform as a dependency for SAMPLE_KEYPHRASE" and a second line "This allows i.MX based platforms to use the samples for the keyphrase detector".

**Comment by xiulipan** [comment]
test this please

File: tools/topology/CMakeLists.txt [hunk 6]
```diff
@@ -119,7 +119,7 @@ set(TPLGS
...
 	"sof-imx8-cs42888-mixer\;sof-imx8-cs42888-mixer"
 	"sof-imx8mp-wm8960-mixer\;sof-imx8mp-wm8960-mixer"
-	"sof-imx8mp-keyword-detect\;sof-imx8mp-keyword-detect"
+	"sof-imx8mp-wm8960-kwd\;sof-imx8mp-wm8960-kwd"
```
Comment (paulstelian97) [inline tools/topology/CMakeLists.txt:122]: Please squash this patch with the first one.

**Review by paulstelian97 — CHANGES_REQUESTED** [review]

**Review by paulstelian97 — APPROVED** [review]

**Comment by lgirdwood** [comment]
CI seems blocked - re start it.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@cristinailie please check your mail - I've added you to developers so CI should automatically run.

**Review by dbaluta — APPROVED** [review]

**Comment by dbaluta** [comment]
SOFCI TEST

**Comment by dbaluta** [comment]
@xiulipan  @zrombel can you guys check the Intel CI?

**Comment by lgirdwood** [comment]
test this please

**Comment by dbaluta** [comment]
@lgirdwood is 'test this please' new magic word for CI to start?

**Comment by dbaluta** [comment]
test this please

**Comment by lgirdwood** [comment]
> @lgirdwood is 'test this please' new magic word for CI to start?

No, it's only for testing PRs for non SOF devlopers. I've added @cristinailie so this wont happens for new PRs.
